src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/BidirectionStreamingTest.cs

```diff
@@ -143,6 +143,8 @@ public async Task AfterReadResponseServerError_ClientWrite()
                 // Server sends RST_STREAM.
                 await connection.WriteFrameAsync(new RstStreamFrame(FrameFlags.EndStream, 0, streamId));
 
+                await Task.Delay(15);
```

Comment:
If this "fixes" it, it's unlikely to be a robust fix.  We generally try not to use delays like this as a way of making tests more robust... something takes just a bit longer when CI is heavily loaded, and it fails again.

Reply:
This seems like bad test. When the frame is written, there is no guarantee IMHO that it is processed before the `WriteAsync` is executed. I agree with @stephentoub that we tend to avoid fixes like this. Putting hardcoded number is always tricky and may or may not work for slow platforms or jitstress.  

It would make sense to me to verify that read throws e.g. the RST was processed before verifying that Write throws. Any thought on this @JamesNK ? (this is test you added in (https://github.com/dotnet/runtime/pull/51094)

Reply:
The test is verifying that the client will throw an error if the server has sent RST_STREAM.
The problem is we don't know when the client has received and processed the RST_STREAM the server has sent.

I don't know if there is a way to get information from WinHttpHandler about whether it has processed the RST_STREAM. My contribution here was small and years ago.

One simple idea to make the test more robust is to keep trying WriteAsync until it fails. e.g.

```cs
// client WriteAsync will fail after WinHttpHandler receives and processes the RST_STREAM
await Assert.ThrowsAsync<IOException>(async () =>
{
    for (int i = 0; i < 10; i++)
    {
        await Task.Delay(50);
        await requestStream.WriteAsync(new byte[50]);
    }
});
```

It's a bit hacky, but WriteAsync will fail eventually, and the delay isn't too large.

If we don't like this fix then a WinHttpHandler expert will need to look at its internals to find something more elegant.

Reply:
> If this "fixes" it, it's unlikely to be a robust fix. We generally try not to use delays like this as a way of making tests more robust... something takes just a bit longer when CI is heavily loaded, and it fails again.

AFAIK, there is no way to get information about `if the client processed reset stream` that's why I thought the best option would be adding delay. But @JamesNK's solution looks much better. Otherwise as also he mentioned, we need to look into WinHttpHandler internals to find some more reliable.

Reply:
> If we don't like this fix then a WinHttpHandler expert will need to look at its internals to find something more elegant.

I don't think this is worth sending emails to the WinHttp team, IMO the proposal in https://github.com/dotnet/runtime/pull/90824#discussion_r1298927095 is good enough.

Reply:
That make sense to me. We have similar patten elsewhere were we try in increments and there is overall limit for failure. That is much better IMHO than adding long delay where we insure the cost every time and it is same for all platforms. 
I would probably make the delay short and add more iterations so in typical case this runs quick.